src/Renci.SshNet/Security/Cryptography/Ciphers/AesCipher.cs

```diff
@@ -77,400 +61,68 @@ public AesCipher(byte[] key, bool pkcs7Padding = false)
         /// <exception cref="ArgumentNullException"><paramref name="key"/> is <see langword="null"/>.</exception>
         /// <exception cref="ArgumentException">Keysize is not valid for this algorithm.</exception>
         public AesCipher(byte[] key, byte[] iv, AesCipherMode mode, bool pkcs7Padding = false)
-            : base(key, BLOCKSIZE, mode: null, padding: pkcs7Padding ? new PKCS7Padding() : null)
+            : base(key, 16, mode: null, padding: null)
         {
-            var keySize = key.Length * 8;
-
-            if (keySize is not (256 or 192 or 128))
```

Comment:
Why not leave this check in place?

Reply:
The `S.S.C.Aes` object will perform the same check